src/billing/List/GCPNewBillingProjectModal.tsx

```diff
@@ -0,0 +1,123 @@
+  onSuccess: (string) => void;
+  onDismiss: () => void;
+  loadAccounts: () => void;
+}
+
+export const GCPNewBillingProjectModal = (props: GCPNewBillingProjectModalProps) => {
+  const [billingProjectName, setBillingProjectName] = useState('');
+  const [existing, setExisting] = useState<string[]>([]);
+  const [isBusy, setIsBusy] = useState(false);
+  const [chosenBillingAccount, setChosenBillingAccount] = useState<GoogleBillingAccount>();
+
+  const submit = _.flow(
+    reportErrorAndRethrow('Error creating billing project'),
+    Utils.withBusyState(setIsBusy)
+  )(async () => {
+    try {
+      // Submit is only enabled when choseBillingAccount is non-null.
+      await Ajax().Billing.createGCPProject(billingProjectName, chosenBillingAccount!.accountName);
+      props.onSuccess(billingProjectName);
+    } catch (error: any) {
+      if (error?.status === 409) {
+        setExisting(_.concat(billingProjectName, existing));
+      } else {
+        throw error;
+      }
+    }
+  });
+
+  const errors = validate({ billingProjectName }, { billingProjectName: billingProjectNameValidator(existing) });
+  const billingLoadedAndEmpty = props.billingAccounts && _.isEmpty(props.billingAccounts);
+  const billingPresent = !_.isEmpty(props.billingAccounts);
+
+  return (
+    <Modal
+      onDismiss={props.onDismiss}
+      shouldCloseOnOverlayClick={false}
+      title='Create Terra Billing Project'
+      showCancel={!billingLoadedAndEmpty}
+      showButtons={!!props.billingAccounts}
+      okButton={
+        billingPresent ? (
+          <ButtonPrimary
+            disabled={errors || !chosenBillingAccount || !chosenBillingAccount.firecloudHasAccess}
+            onClick={submit}
+          >
+            Create
+          </ButtonPrimary>
+        ) : (
+          <ButtonPrimary onClick={props.onDismiss}>Ok</ButtonPrimary>
+        )
+      }
+    >
+      {billingLoadedAndEmpty && (
+        <>
+          {"You don't have access to any billing accounts.  "}
+          <Link href='https://support.terra.bio/hc/en-us/articles/360026182251' {...Utils.newTabLinkProps}>
+            Learn how to create a billing account.
+            <Icon icon='pop-out' size={12} style={{ marginLeft: '0.5rem' }} />
+          </Link>
+        </>
+      )}
+      {billingPresent && (
+        <>
+          <CreateGCPBillingProject
+            billingAccounts={props.billingAccounts}
+            chosenBillingAccount={chosenBillingAccount}
+            setChosenBillingAccount={setChosenBillingAccount}
+            billingProjectName={billingProjectName}
+            setBillingProjectName={setBillingProjectName}
+            existing={existing}
+          />
+          {!!chosenBillingAccount && !chosenBillingAccount.firecloudHasAccess && (
+            <div style={{ fontWeight: 500, fontSize: 13 }}>
+              <div style={{ margin: '0.25rem 0 0.25rem 0', color: colors.danger() }}>
+                Terra does not have access to this account.{' '}
```

Comment:
Hacked locally to show this text:

![image](https://github.com/user-attachments/assets/40f6d468-010c-461b-8b75-e3f26e8ee627)